From 9a2ab5b74639083191c89d34bc6823d1367c5084 Mon Sep 17 00:00:00 2001
From: uros <uros@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Sun, 9 Jul 2017 21:01:42 +0000
Subject: [PATCH] Update gcc.dg/stack-layout-dynamic-1.c

	PR target/81313
	* gcc.dg/stack-layout-dynamic-1.c (bar): Add 4 additional
	integer argumets to bypass x86_64 outgoing args optimization.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@250086 138bc75d-0d04-0410-961f-82ee72b054a4

(cherry picked from commit ac2f5a408fa8768a0936b256f5cd7e0fe9f3b963)
---
 gcc/testsuite/gcc.dg/stack-layout-dynamic-1.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gcc/testsuite/gcc.dg/stack-layout-dynamic-1.c b/gcc/testsuite/gcc.dg/stack-layout-dynamic-1.c
index 9f2d37d..7e59f28 100644
--- a/gcc/testsuite/gcc.dg/stack-layout-dynamic-1.c
+++ b/gcc/testsuite/gcc.dg/stack-layout-dynamic-1.c
@@ -1,15 +1,15 @@
-/* Verify that run time aligned local variables are aloocated in the prologue
+/* Verify that run time aligned local variables are allocated in the prologue
    in one pass together with normal local variables.  */
 /* { dg-do compile } */
 /* { dg-options "-O0 -fomit-frame-pointer" } */
 /* { dg-require-effective-target ptr32plus } */
 
-extern void bar (void *, void *, void *);
+extern void bar (void *, void *, void *, void *, void *, void *, void *);
 void foo (void)
 {
-  int i;
+  int i, j, k, l, m;
   __attribute__ ((aligned(65536))) char runtime_aligned_1[512];
   __attribute__ ((aligned(32768))) char runtime_aligned_2[1024];
-  bar (&i, &runtime_aligned_1, &runtime_aligned_2);
+  bar (&i, &j, &k, &l, &m, &runtime_aligned_1, &runtime_aligned_2);
 }
 /* { dg-final { scan-assembler-not "cfi_def_cfa_register" } } */
-- 
2.9.3

